app/src/main/java/com/eveningoutpost/dexdrip/receivers/aidex/AidexReceiver.java

```diff
@@ -0,0 +1,311 @@
+                            } break;
+
+                            default:
+                                UserError.Log.e(TAG, "Unknown action! " + action);
+                                break;
+                        }
+                    } finally {
+                        JoH.benchmark("Aidex process");
+                        JoH.releaseWakeLock(wl);
+                    }
+                } // lock
+            }
+
+        }.start();
+
+    }
+
+
+    private void processNewBGEstimate(Bundle bundle) {
+
+        if (!bundle.containsKey(AIDEX_TIMESTAMP) ||
+                !bundle.containsKey(AIDEX_SENSOR_ID) ||
+                !bundle.containsKey(AIDEX_BG_TYPE) ||
+                !bundle.containsKey(AIDEX_BG_VALUE)
+        ) {
+            UserError.Log.e(TAG, "Aidex Broadcast for NewBGEstimate invalid, missing one of required parameters: SendorId, Timestamp, BgType or BgValue! Ignoring broadcast.");
+            return;
+        }
+
+        String sensorId = bundle.getString(AIDEX_SENSOR_ID);
+        long timeStamp = bundle.getLong(AIDEX_TIMESTAMP);
+        String bgType = bundle.getString(AIDEX_BG_TYPE);
+        Double bgValue = bundle.getDouble(AIDEX_BG_VALUE);
+
+        int bgValueMgDl = 0;
+
+        if (AidexBroadcastIntents.UNIT_MG_DL.equalsIgnoreCase(bgType)) {
+            bgValueMgDl = bgValue.intValue();
+        } else if (AidexBroadcastIntents.UNIT_MMOL_L.equalsIgnoreCase(bgType)) {
+            bgValueMgDl = (int)(bgValue * Constants.MMOLL_TO_MGDL);
+        } else {
+            UserError.Log.e(TAG, "Aidex Broadcast BgType invalid. BgType needs to be either: mg/dl or mmol/l.");
+            return;
+        }
+
+        if (bgValueMgDl <= 0) {
+            UserError.Log.w(TAG, "Aidex Broadcast received BG=0, ignoring it.");
+            return;
+        }
+
+        checkIfCorrectSensorIsRunning(sensorId, timeStamp);
+
+        UserError.Log.i(TAG, "Aidex Broadcast NewBGEstimate received: bg=" + bgValueMgDl + ", time=" + JoH.dateTimeText(timeStamp));
+        BgReading.bgReadingInsertFromInt(bgValueMgDl, timeStamp, segmentation_timeslice, false);
+
+    }
+
+
+    private void checkIfCorrectSensorIsRunning(String sensorId, long timeStamp) {
+        Sensor currentSensor = Sensor.currentSensor();
+        if(currentSensor!=null) {
+            if (!currentSensor.uuid.equals(sensorId)) {
+                Sensor.stopSensor();
+                Sensor.create(timeStamp - 10000, sensorId);
+            }
+        } else {
+            Sensor.create(timeStamp - 10000, sensorId);
+        }
+    }
+
+
+    private void processCalibration(Bundle bundle) {
+        if (Pref.getBooleanDefaultFalse("accept_broadcast_calibrations")) {
+
+            final long calibration_timestamp = bundle.getLong(AIDEX_TIMESTAMP, -1);
+            double bgValue = bundle.getDouble(AIDEX_BG_VALUE, -1);
+            final String units = bundle.getString(AIDEX_BG_TYPE, "mg/dl");
```

Comment:
Here you can use `AidexBroadcastIntents.UNIT_MG_DL` as well.

Reply:
Fixed.